internal/controller/nginx/config/servers.go

```diff
@@ -318,16 +353,59 @@ it was the final location that had -inference in the name.
 location /coffee {
 	js_content httpmatches.match; // chooses backend and redirects to appropriate internal inference location
 }
-location /_ngf-internal-rule0-route0-inference {
+location /_ngf-internal-upstreamNamePrimary-routeNS-routeName-rule0 {
 	internal;
 
-	set $epp_internal_path /_ngf-internal-rule0-route0;
-	js_content epp.getEndpoint; // redirects to /_ngf-internal-rule0-route0
+	set $epp_internal_path /_ngf-internal-proxy-pass-rule0-route0-backend0-inference;
+	js_content epp.getEndpoint; // redirects to /_ngf-internal-proxy-pass-rule0-route0-backend0-inference
 }
-location /_ngf-internal-rule0-route0 {
+location /_ngf-internal-proxy-pass-rule0-route0-backend0-inference {
 	internal;
 	proxy_pass http://$inference-backend;
 }
+---------------
+Inference extension with multiple backends with HTTP matching conditions.
+
+External location calls httpmatch NJS module. The module determines the HTTP request conditions that exist
+and which backend to use, then redirects to an internal location which will rewrite to another internal inference
+location based on a split clients variable. That internal inference location calls the inference NJS module
+to get the AI endpoint to proxy to, then redirects to the internal location that proxies to the backend.
```

Comment:
I wish this was less insane lol, but unfortunately it seems necessary right now.